wgpu-core/src/instance.rs

```diff
@@ -50,27 +50,36 @@ pub struct Instance {
     ))]
     pub vulkan: Option<gfx_backend_vulkan::Instance>,
     #[cfg(any(target_os = "ios", target_os = "macos"))]
-    pub metal: gfx_backend_metal::Instance,
+    pub metal: Option<gfx_backend_metal::Instance>,
     #[cfg(windows)]
     pub dx12: Option<gfx_backend_dx12::Instance>,
     #[cfg(windows)]
-    pub dx11: gfx_backend_dx11::Instance,
+    pub dx11: Option<gfx_backend_dx11::Instance>,
 }
 
 impl Instance {
-    pub fn new(name: &str, version: u32) -> Self {
+    pub fn new(name: &str, version: u32, backends: BackendBit) -> Self {
+        let inputs = AdapterInputs::Mask(backends, |_| ());
```

Comment:
we don't need this. Instead, each constructor would check `if backends.contains(...) { Some(...) } else {None}`